scala-diff/shared/src/main/scala/munit/diff/console/Printers.scala

```diff
@@ -0,0 +1,233 @@
+// Adaptation of https://github.com/lihaoyi/PPrint/blob/e6a918c259ed7ae1998bbf58c360334a3f0157ca/pprint/src/pprint/Walker.scala
+package munit.diff.console
+
+import munit.diff.{EmptyPrinter, Printable, Printer}
+
+import scala.annotation.switch
+import munit.diff.Clues
+import munit.diff.internal.Compat
+
+object Printers {
+
+  /** Pretty-prints the value in a format that's optimized for producing diffs */
+  def print(any: Any, printer: Printer = EmptyPrinter): String = {
+    var height = printer.height
+    val out = new StringBuilder()
+    val indentStep = 2
+    def loop(a: Any, indent: Int): Unit = {
+      height -= 1
+      if (height < 0) {
+        out.append("...")
+        return
+      }
+      val nextIndent = indent + indentStep
+      val isDone = printer.print(a, out, indent)
+      if (!isDone) {
+        a match {
+          case null         => out.append("null")
+          case x: Printable => x.print(out, indent)
+          case x: Char =>
+            out.append('\'')
+            if (x == '\'') out.append("\\'")
+            else printChar(x, out)
+            out.append('\'')
+          case x: Byte   => out.append(x.toString())
+          case x: Short  => out.append(x.toString())
+          case x: Int    => out.append(x.toString())
+          case x: Long   => out.append(x.toString())
+          case x: Float  => out.append(x.toString())
+          case x: Double => out.append(x.toString())
+          case x: String => printString(x, out, printer)
+          case x: Clues =>
```

Comment:
Could we just make Printer a class and add support for Clues in a child class?

Or maybe a custom class like:
```scala
trait CustomPrintable{
  def append(out: StringBuilder)
}
```
and make Clue implement it?